### PR TITLE
Show notice for non existent controller property.

### PR DESCRIPTION
**Pull request by ADmad**
Refs https://github.com/cakephp/cakephp/issues/12590#issuecomment-423932380

### CODE REVIEW

File: src/Controller/Controller.php [hunk 1]
```diff
@@ -371,6 +371,19 @@ public function __get($name)
 
         list($plugin, $class) = pluginSplit($this->modelClass, true);
         if ($class !== $name) {
```
Comment (garas) [inline src/Controller/Controller.php]: Maybe reverse `if` and explain:

```
if ($class === $name) {
   // Autoload table if $name matches controller name
   return $this->loadModel($plugin . $class);
}

// notice

return false; // remove or explicitly return null at 4.x
```

**Comment by markstory** [comment]
Looks like we have a few undefined properties being used by our tests :cry:

**Comment by ADmad** [comment]
Not really, the error doesn't make sense. It's complaining about missing `Controller::$viewClass` but that property is available through the `ViewVarsTrait`.

**Comment by garas** [comment]
`Controller::$viewClass` is defined in `ViewVarsTrait`, but it is `unset()` after first use.

> https://github.com/cakephp/cakephp/pull/12035 I've added an unset() for viewClass to avoid a permanent deprecation warning when RequestHandlerComponent is used.

https://github.com/cakephp/cakephp/blob/13ab4d788ce553056938b60b3039197b8edd9993/src/View/ViewVarsTrait.php#L75-L78

If second `render()` tries access value, it is no more available. It should probably leave property set.

**Comment by ADmad** [comment]
`PHPUnit\Framework\Error\Notice` is not available in phpunit 5.7 :slightly_frowning_face:

**Comment by ADmad** [comment]
Meh, took care of it with a `class_exists()` check.

**Pull request merged**